astroquery/utils/tap/model/modelutils.py

```diff
@@ -36,11 +37,25 @@ def read_results_table_from_file(file_name, output_format, *, correct_units=True
 
     if check_file_exists(file_name):
 
-        result = APTable.read(file_name, format=astropy_format)
+        if output_format == 'json':
+            files = {}
+            with open(file_name) as f:
+                data = json.load(f)
 
-        if correct_units:
-            utils.modify_unrecognized_table_units(result)
+                if data.get('data') and data.get('metadata'):
 
-        return result
+                    files['data'] = APTable.read(json.dumps({"data": data['data']}), format=astropy_format)
+                    files['metadata'] = APTable.read(json.dumps({"metadata": data['metadata']}), format=astropy_format)
+
+                    return files
+                else:
+                    return APTable.read(file_name, format=astropy_format)
+        else:
+            result = APTable.read(file_name, format=astropy_format)
+
+            if correct_units:
```

Comment:
are units correct for the JSON output?

Reply:
Hi @bsipocz, in order to correct the units in the astropy Table, we have changed the implementation: a Table is build from scratch based on the returned json file and the units are corrected. The original json is directly saved into the file system if the parameters dump_to_file= True and output_file are set.